django_declarative_apis/machinery/tasks.py

```diff
@@ -276,6 +286,145 @@ def schedule_future_task_runner(
                     raise err
 
 
+@celery_task(
+    ignore_results=True,
+    time_limit=getattr(settings, "DDA_DEFERRED_TASK_TIME_LIMIT", 999999),
+    soft_time_limit=getattr(settings, "DDA_DEFERRED_TASK_SOFT_TIME_LIMIT", 999999),
+)
+def generic_future_task_runner(
+    endpoint_class_name,
+    endpoint_method_name,
+    packed_args,
+    packer_class_name,
+    task_job_count=0,
+    task_creation_time=None,
+    scheduled_execution_delay=0,
+    retry_params=None,
+    correlation_id=None,
+):
+    endpoint_class = locate(endpoint_class_name)
+    endpoint_task = getattr(endpoint_class, endpoint_method_name)
+    packer_class = locate(packer_class_name)
+    args, kwargs = packer_class.unpack(packed_args)
+
+    _set_correlation_id(correlation_id)
+
+    _log_task_stats(
+        endpoint_method_name,
+        None,
+        scheduled_execution_delay,
+        task_creation_time,
+        task_job_count,
+        correlation_id,
+    )
+
+    logger.info(
+        "future_task_runner: method=%s, resource_id=%s",
+        endpoint_method_name,
+        None,
+    )
+
+    try:
+        endpoint_task.task_runner(*args, **kwargs)
+    except Exception as e:
+        if retry_params is None:
+            raise
+
+        retry_params = RetryParams(*retry_params)
+        if retry_params.retry_exception_filter and not any(
+            [
+                f"{e.__class__.__module__}.{e.__class__.__name__}" == ex_type
+                for ex_type in retry_params.retry_exception_filter
+            ]
+        ):
+            # this exception is not retryable
+            raise
+
+        if retry_params.retries_remaining == 0:
+            # out of chances
+            raise
+
+        _log_retry_stats(endpoint_method_name, resource_instance_id, correlation_id)
+        task_runner_args = (
+            endpoint_class_name,
+            endpoint_method_name,
+            packed_args,
+            packer_class_name,
+        )
+        task_runner_kwargs = {
+            "task_creation_time": time.time(),
+            "scheduled_execution_delay": scheduled_execution_delay,
+            "task_args": (args, kwargs),
+        }
+        schedule_generic_future_task_runner(
+            task_runner_args,
+            task_runner_kwargs,
+            retries=retry_params.retries_remaining - 1,
+            countdown=retry_params.countdown * 2 or 1,
+            queue=retry_params.queue,
+            routing_key=retry_params.routing_key,
+        )
+
+
+def schedule_generic_future_task_runner(
```

Comment:
This looks identical to `schedule_future_task_runner` unless I'm missing something.